framework/src/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -1512,9 +1509,6 @@ void setLowerBoundField(AnnotatedTypeMirror type) {
          * @return the lower bound field.
          */
         public AnnotatedTypeMirror getLowerBoundField() {
-            if (lowerBound != null) {
```

Comment:
So I believe there might be a problem with this approach.  Currently, the annotations list in this class is protected which is bad.  It should be private.  So, if a class that has access to the annotations adds a primary annotation type variable through this mechanism, it won't add them to the bounds.  So if fixupBoundAnnotations is not called when getLowerBoundField/getLowerBound and getUpperBoundField/getUpperBound then the methods won't return the correct type.